plugins/out_azure_logs_ingestion/azure_logs_ingestion_conf.c

```diff
@@ -54,21 +53,45 @@ struct flb_az_li* flb_az_li_ctx_create(struct flb_output_instance *ins,
         return NULL;
     }
 
-    /* config: 'client_id' */
-    if (!ctx->client_id) {
-        flb_plg_error(ins, "property 'client_id' is not defined");
-        flb_az_li_ctx_destroy(ctx);
-        return NULL;
-    }
-    /* config: 'tenant_id' */
-    if (!ctx->tenant_id) {
-        flb_plg_error(ins, "property 'tenant_id' is not defined");
-        flb_az_li_ctx_destroy(ctx);
-        return NULL;
-    }
-    /* config: 'client_secret' */
-    if (!ctx->client_secret) {
-        flb_plg_error(ins, "property 'client_secret' is not defined");
+    /* Auth method validation and setup */
+    if (!ctx->auth_type_str || strlen(ctx->auth_type_str) == 0) {
```

Comment:
These if-else block (L57-L67) can be written as one like so (remove redundancy):

```c
if (!ctx->auth_type_str || strlen(ctx->auth_type_str) == 0 || strcasecmp(ctx->auth_type_str, "service_principal") == 0) {
        /* Default to service_principal if auth_type_str is NULL or empty */
        ctx->auth_type = FLB_AZ_LI_AUTH_SERVICE_PRINCIPAL;

        /* Verify required parameters for Service Principal auth */
        if (!ctx->tenant_id || !ctx->client_id || !ctx->client_secret) {
            flb_plg_error(ins, "When using service_principal auth, tenant_id, client_id, and client_secret are required");
            flb_az_li_ctx_destroy(ctx);
            return NULL;
        }
    }
```

Reply:
Makes sense, merged the two in https://github.com/fluent/fluent-bit/pull/10867/commits/a2c50ed2229d5c468bfbfac2d80464d5c1f83888